src/components/table/index.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+
+import { Table, TableHead, TableHeaderCell, TableBody, TableRow, TableCell } from "@tremor/react"
+import { useEffect, useState } from "react";
+import type { ColumnDefinition, TableState } from "@/types/tableTypes";
+
+export default function CatchmentTable({
+  fetchItems,
+  columnDefinitions,
+  tableState,
+  setTableState,
+  count,
+}: {
+  fetchItems: () => Promise<any[]>;
+  columnDefinitions: ColumnDefinition[];
+  tableState: TableState;
+  setTableState: (tableState: TableState) => void;
+  count: () => Promise<number>;
+}) {
+  const [items, setItems] = useState<any[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    fetchData();
+  }, [tableState.page]);
+
+  useEffect(() => {
+    count().then((totalCounted) => {
+      setTableState({ ...tableState, total: totalCounted });
+    });
+  }, []);
+
+  const fetchData = async () => {
+    setIsLoading(true);
+    const response = await fetchItems();
+    setItems(response);
+    setIsLoading(false);
+  }
+
+  const itterateThroughNestedAttrs = (object: Record<string, any>, keys: string): any => {

```

Comment:
Tohle je hůř čitelné, proč to tam je (i když jsem to nakonec pochopil), zasloužilo by si to komentář (stačí jeden řádek, že to umožňuje číst "obj.propObj.prop"). A nebo to radši úplně nahradit funkcí `row => row.propObj.prop`. Protože teď když přejmenuješ prop v entitě, tak ve stringu ti zůstane starý název.

Reply:
Myslel jsem, že ten název té metody je "self explanatory", ale klidně tam hodím nějaký komentář.
Úplně nerozumím té druhé části o tom nahrazení. Možná se o tom můžeme pobavit až přijedeš.

Reply:
Myslel jsem dát do ColumnDefinition místo key - stringu s názvem cesty k property (jako je "propObj.prop"), tam posílat callback, který vezme celý otypovaný row (s generickým typem T) a vrátí tu hodnotu - díky tomu můžeš vracet i nějaké upravené / přepočtené hodnoty, a zároveň máš kontrolu od IDE ohledně refactoringu, protože budeš mít "živé" properties přesně toho tvého typu, nebude to zakódované do stringu.

Něco takového:
```
export type ColumnDefinition<T> = {
  title: string;
  getValue: T => any; // (případně vracet string)
}
```

Reply:
Jo, rozumím. To dává smysl...

Reply:
Přetvořeno...